src/qibo/backends/abstract.py

```diff
@@ -318,12 +318,16 @@ def calculate_overlap_density_matrix(self, state1, state2):  # pragma: no cover
         raise_error(NotImplementedError)
 
     @abc.abstractmethod
-    def calculate_eigenvalues(self, matrix, k=6):  # pragma: no cover
+    def calculate_eigenvalues(
+        self, matrix, k: int = 6, hermitian: bool = True
+    ):  # pragma: no cover
```

Comment:
Pay attention: this kind of change affects the public API, and it could be breaking.

In particular, this is mostly not breaking because of the default value (and also because almost everyone subclasses the `NumpyBackend`...), but if there were backends directly subclassing this, respecting its signature, they would now be broken, since they would not support the extra argument (and they would be flagged by linters, e.g. Pylint in the CI).

So, strictly speaking, it is breaking, but by virtue of the `NumpyBackend`-subclassing habit it should be irrelevant.

Reply:
In this case, I'm not adding anything new everywhere, just fixing an inconsistency in the method in the `abstract` class (method that is being overwritten by every backend anyways)

Reply:
Yes, that's way, I added the note about it being irrelevant.

Moreover, Qibo is not that stable anyway, also because the public API is not well-identified.
But, in principle, it's the `AbstractBackend` defining the interface, not the `NumpyBackend`.

I just wanted to flag it, just because I'd like to move towards a more stable package, and a more consistent use of semver (as a guarantee to the Qibo users).
But I agree with you in qualifying this as mostly a bugfix (but if Qibo was more stable and widely used, we should have considered this "bug" as the current interface, to avoid breaking users' code - no matter if it was like that by mistake).

Reply:
It is also a bug fix because that is making some tests fail for `cupy` / `cuquantum`. The main reason for this fix, actually. But yeah we are in agreement.